galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+        @param theta_x  Field angle corresponding to center of output array.
+        @param theta_y  Ditto.
+        @returns   Array of pathlength differences in nanometers.  Multiply by 2pi/wavelength to get
+                   array of phase differences.
+        """
+        # This should return an nx-by-nx pixel array with scale `scale` (in meters) representing the
+        # effective difference in path length (nanometers) for rays originating from different
+        # points in the pupil plane.  The `theta_x` and `theta_y` params indicate the position on
+        # the focal plane, or equivalently the position on the sky from which the rays originate.
+        raise NotImplementedError
+
+    def pupil_scale(self, lam, scale_unit=galsim.arcsec):
+        """Compute a good pupil_scale in meters for this atmospheric layer.
+
+        @param lam        Wavelength in nanometers.
+        @param scale_unit Scale unit assumed when eventually image.  [Default: galsim.arcsec]
+        @returns  Good pupil scale size in meters.
+        """
+        raise NotImplementedError
+
+
+class AtmosphericScreen(PhaseScreen):
+    """ Abstract base class for an atmospheric phase screen.  Not intended to be instantiated
+    directly.  Holds methods common to concrete subclasses.
+
+    @param screen_size   Physical extent of square phase screen in meters.  This should be large
+                         enough to accommodate the desired field-of-view of the telescope as well as
+                         the meta-pupil defined by the wind velocity and exposure time.  Note that
+                         the screen will have periodic boundary conditions, so it's technically
+                         possible to use a smaller sized screen than technically necessary, though
+                         this may introduce artifacts into PSFs or PSF correlations functions.
+                         Note that screen_size may be tweaked by the initializer to ensure
+                         screen_size is a multiple of screen_scale.
+    @param screen_scale  Physical pixel scale of phase screen in meters.  A fraction of the Fried
+                         parameter is usually sufficiently small, but users should test the effects
+                         of this parameter to ensure robust results. [Default: half of r0_500]
+    @param altitude      Altitude of phase screen in km.  This is with respect to the telescope, not
+                         sea-level.  [Default: 0.0]
+    @param time_step     Interval to use when advancing the screen in time in seconds.
+                         [Default: 0.03]
+    @param r0_500        Fried parameter setting the amplitude of turbulence; contributes to "size"
+                         of the resulting atmospheric PSF.  Specified at wavelength 500 nm, in units
+                         of meters.  [Default: 0.2]
+    @param L0_inv        Inverse outer scale in inverse meters, beyond which turbulence power
```

Comment:
This seems more complicated than necessary.  Couldn't we just have the user specify L0 in meters?